src/libstd/ffi/c_str.rs

```diff
@@ -328,6 +350,53 @@ impl CStr {
     pub fn to_bytes_with_nul(&self) -> &[u8] {
         unsafe { mem::transmute::<&[libc::c_char], &[u8]>(&self.inner) }
     }
+
+    /// Yields a `&str` slice if the `CStr` contains valid UTF-8.
+    ///
+    /// This function will calculate the length of this string and check for
+    /// UTF-8 validity, and then return the `&str` if it's valid.
+    ///
+    /// > **Note**: This method is currently implemented to check for validity
+    /// > after a 0-cost cast, but it is planned to alter its definition in the
+    /// > future to perform the length calculation in addition to the UTF-8
+    /// > check whenever this method is called.
+    #[unstable(feature = "cstr", reason = "recently added")]
```

Comment:
I agree with @steveklabnik that we're trying to cut down on feature names. In this case `cstr` is somewhat ok because there's not an already existing blanket `cstr` feature, but I'd personally prefer that these new APIs be added as `cstr_to_str` for example.